src/procedures/__tests__/SetDividendsWallet.ts

```diff
@@ -0,0 +1,152 @@
+      'MockedTokenFactoryObject'
+    );
+    securityTokenMock = ImportMock.mockClass(
+      contractWrappersModule,
+      'SecurityToken_3_0_0'
+    );
+
+    erc20DividendMock = ImportMock.mockClass(
+      contractWrappersModule,
+      'ERC20DividendCheckpointContract_3_0_0'
+    );
+
+    tokenFactoryMock.mock(
+      'getSecurityTokenInstanceFromTicker',
+      securityTokenMock.getMockInstance()
+    );
+
+    contextMock.set('contractWrappers', wrappersMock.getMockInstance());
+    wrappersMock.set('tokenFactory', tokenFactoryMock.getMockInstance());
+  });
+
+  afterEach(() => {
+    restore();
+  });
+
+  describe('Types', () => {
+    test('should extend procedure and have SetDividendsWallet type', async () => {
+      target = new SetDividendsWallet(
+        { ...params, dividendType: DividendType.Erc20 },
+        contextMock.getMockInstance()
+      );
+      expect(target instanceof Procedure).toBe(true);
+      expect(target.type).toBe(ProcedureType.SetDividendsWallet);
+    });
+  });
+
+  describe('SetDividendsWallet', () => {
+    test('should throw if there is no valid dividend type being provided', async () => {
+      // Instantiate SetDividendsWallet with incorrect dividend type
+      target = new SetDividendsWallet(
+        {
+          symbol: params.symbol,
+          dividendType: 'wrong' as DividendType,
+          address: params.address,
+        },
+        contextMock.getMockInstance()
+      );
+
+      expect(target.prepareTransactions()).rejects.toThrow(
+        new PolymathError({
+          code: ErrorCode.ProcedureValidationError,
+          message: "Dividends of the specified type haven't been enabled",
+        })
+      );
+    });
+
+    test('should throw if an Erc20 dividend module is not attached', async () => {
+      target = new SetDividendsWallet(
+        { ...params, dividendType: DividendType.Erc20 },
+        contextMock.getMockInstance()
+      );
+
+      wrappersMock.mock('getAttachedModules', Promise.resolve([]));
+
+      // Real call
+      expect(target.prepareTransactions()).rejects.toThrowError(
+        new PolymathError({
+          code: ErrorCode.ProcedureValidationError,
+          message: "Dividends of the specified type haven't been enabled",
+        })
+      );
+    });
+
+    test('should throw if an Eth dividend module is not attached', async () => {
+      target = new SetDividendsWallet(
+        { ...params, dividendType: DividendType.Eth },
+        contextMock.getMockInstance()
+      );
+
+      wrappersMock.mock('getAttachedModules', Promise.resolve([]));
+
+      // Real call
+      expect(target.prepareTransactions()).rejects.toThrowError(
+        new PolymathError({
+          code: ErrorCode.ProcedureValidationError,
+          message: "Dividends of the specified type haven't been enabled",
+        })
+      );
+    });
+
+    test('should add transactions to the queue for change wallet with a new Erc20 dividend module', async () => {
```

Comment:
Fede, your code is good, nice work. The only comment I have is that I feel like you should write this test two times. I think you should test both 

"should add transactions to the queue for change wallet with a new **Erc20** dividend module"
"should add transactions to the queue for change wallet with a new **ETH** dividend module"

Reply:
Agreed

Reply:
The description text is a bit off though. Should be

"should add a transaction to the queue to change the dividends wallet of an attached ERC20 dividends module"

And the same for ETH